relay-server/src/envelope.rs

```diff
@@ -871,6 +872,27 @@ impl Item {
         self.headers.other.insert(name.into(), value.into())
     }
 
+    /// Counts how many spans are contained in a transaction payload.
+    ///
+    /// The transaction itself represents a span as well, so this function returns
+    /// `len(event.spans) + 1`.
+    ///
+    /// Returns zero if
+    /// - the item is not a transaction,
+    /// - the spans have already been extracted (in which case they are represented elsewhere).
+    pub fn count_nested_spans(&self) -> usize {
+        #[derive(Debug, Deserialize)]
+        struct PartialEvent {
+            spans: SeqCount,
+        }
+
+        if self.ty() != &ItemType::Transaction || self.spans_extracted() {
+            return 0;
+        }
+
+        serde_json::from_slice::<PartialEvent>(&self.payload()).map_or(0, |event| event.spans.0 + 1)
```

Comment:
Should we persist this in a header and/or memoize it?

Reply:
We could if necessary, but that would mean we have to update it as well (e.g. trimming processor which removes spans), so I'm hesitant to add it now.